### PR TITLE
express-jwt - enhance callback type

### PR DESCRIPTION
**Pull request by frogcjn**
secretCallback:
https://github.com/auth0/express-jwt/blob/master/lib/index.js#L92

IsRevokedCallback:
https://github.com/auth0/express-jwt/blob/master/README.md#revoked-tokens

GetTokenCallback:
https://github.com/auth0/express-jwt/blob/master/README.md#multi-tenancy

### CODE REVIEW

File: mongoose/mongoose.d.ts [hunk 1]
```diff
@@ -276,8 +276,7 @@ declare module "mongoose" {
 
     pre(method: string, fn: HookSyncCallback, errorCb?: HookErrorCallback): Schema;
     pre(method: string, isAsync: boolean, fn: HookAsyncCallback, errorCb?: HookErrorCallback): Schema;
-    post(method: string, fn: HookSyncCallback, errorCb?: HookErrorCallback): Schema;
-    post(method: string, isAsync: boolean, fn: HookAsyncCallback, errorCb?: HookErrorCallback): Schema;
```
Comment (RyanCavanaugh) [inline mongoose/mongoose.d.ts]: What happened to the overload using `isAsync` ?

File: express-jwt/express-jwt.d.ts [hunk 2]
```diff
@@ -12,26 +12,35 @@ declare module "express-jwt" {
 
     function jwt(options: jwt.Options): jwt.RequestHandler;
 
-    interface IDoneCallback<T> {
-        (err: Error, result: T): void;
-    }
+    namespace jwt {
 
-    type ICallback = <T>(req: express.Request, payload: T, done: IDoneCallback<boolean>) => void;
+        export type secretType = string | Buffer
```
Comment (RyanCavanaugh) [inline express-jwt/express-jwt.d.ts]: I don't understand the naming here -- `secretType` ?

Comment (frogcjn) [inline express-jwt/express-jwt.d.ts]: so how you name it?

File: mongoose/mongoose.d.ts (hunk 1 above)
Comment (frogcjn) [inline mongoose/mongoose.d.ts]: See http://mongoosejs.com/docs/middleware.html#post

and the change in mongoose is already merged, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9850

I don't know why the diff show it again.

File: express-jwt/express-jwt.d.ts (hunk 2 above)
Comment (frogcjn) [inline express-jwt/express-jwt.d.ts]: @RyanCavanaugh so how you name the type of secret? I'll revise it.

File: mongoose/mongoose.d.ts (hunk 1 above)
Comment (frogcjn) [inline mongoose/mongoose.d.ts]: @RyanCavanaugh 
See http://mongoosejs.com/docs/middleware.html#post

post is different than pre. 

and the change in mongoose is already merged, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9850

I don't know why diff shows it again.

File: express-jwt/express-jwt.d.ts (hunk 2 above)
Comment (frogcjn) [inline express-jwt/express-jwt.d.ts]: @RyanCavanaugh need feedback, thanks!

Comment (frogcjn) [inline express-jwt/express-jwt.d.ts]: @johnnyreilly some others could consider to merge this PR?

**Comment by johnnyreilly** [comment]
I think this is probably good enough for now. Thanks!

**Pull request merged**